File: pubnub-gson/pubnub-gson-impl/src/main/java/com/pubnub/internal/endpoints/TimeImpl.java
package com.pubnub.internal.endpoints;

import com.pubnub.api.endpoints.Time;
import com.pubnub.api.models.consumer.PNTimeResult;
import com.pubnub.internal.EndpointInterface;
import com.pubnub.internal.PubNubCore;
import org.jetbrains.annotations.NotNull;

public class TimeImpl extends IdentityMappingEndpoint<PNTimeResult> implements Time {

    public TimeImpl(PubNubCore pubnub) {
        super(pubnub);
    }

    @Override
    @NotNull
    protected EndpointInterface<PNTimeResult> createAction() {
        return pubnub.time();
    }
}

File: pubnub-kotlin/pubnub-kotlin-impl/src/main/kotlin/com/pubnub/internal/endpoints/TimeImpl.kt
package com.pubnub.internal.endpoints

import com.pubnub.api.endpoints.Time
import com.pubnub.api.models.consumer.PNTimeResult
import com.pubnub.internal.EndpointImpl
import com.pubnub.internal.PubNubImpl

/**
 * @see [PubNubImpl.time]
 */
class TimeImpl internal constructor(time: TimeInterface) :
    TimeInterface by time,
    Time,
    EndpointImpl<PNTimeResult>(time)
